src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Xamarin.Android.Build.Tests.csproj

```diff
@@ -4,6 +4,8 @@
     <TargetFrameworks>net472;net6.0</TargetFrameworks>
     <LibZipSharpBundleAllNativeLibraries>true</LibZipSharpBundleAllNativeLibraries>
     <OutputPath>..\..\..\..\bin\Test$(Configuration)</OutputPath>
+    <SignAssembly>true</SignAssembly>
+    <AssemblyOriginatorKeyFile>..\..\..\..\product.snk</AssemblyOriginatorKeyFile>
```

Comment:
I don't think we need to strong name the test projects, right?

Reply:
We do because our tests make use of Internal classes via the `InternalVisibleTo` attribute.

Reply:
I think you'd only need to strong name the tests, if another project referenced the tests?

You should be able to reference strong named assemblies from a project that isn't strong named.

Reply:
Nope, because if the `InternalVisibleTo` works, you need to provide the `PublicKey` as part of the assembly input. Otherwise you get a compiler error. See https://github.com/xamarin/xamarin-android/pull/6116/files#diff-fc3eea7f2d103f214e089b57dc3ae56bb2e18fea95703719084ded82383c7c0cR21. 

To get that PublicKey the assembly needs to be signed.

Reply:
Oh... I see. We actually have a circular dependency here.